### DISCUSSION TITLE
Idea: add support for enums in `swc_fast_ts_strip` by putting all the enums definitions on the first or last line of the file

### DISCUSSION THREAD

**Discussion by remorses** [swc-project/swc#9825] [Ideas]
Currently Node.js type stripping does not support enums because it would need sourcemaps to do so.

What is the enums definitions were added at the top of the file?

swc would then remove the first \n of the file, so only the source mappings of the first line would be wrong, which would contain only imports anyway most of the time.

Here is an example:

Before
```ts
import x from './x'
import y from './y'

enum something {
  x
  y
}

console.log(something.x)
```

After
```ts
var something; (function (something) { something[something["x"] = 0] = "x"; something[something["y"] = 1] = "y"; })(something || (something = {})); import x from './x'
import y from './y





console.log(something.x)
```

@kdy1

**Comment by kdy1** [comment]
cc @marco-ippolito @robpalme Thoughts?

**Comment by robpalme** [comment]
There's a few problems with this.

- Enums can contain expressions that have side-effects and side-causes.  So it's not safe to arbitrarily relocate them.
- Pre-pending the first line means the errors and debugging on the first line will be incorrect.
- Any kind of expansion that offsets the generated code will corrupt all the v8 (c8) code coverage stats which relies on identical positions

I don't believe it's possible to reconcile this.

**Reply by marco-ippolito** [reply]
Agree with Rob. It's not possible to support it safely
